cpp/src/io/parquet/page_enc.cu

```diff
@@ -337,8 +339,13 @@ __global__ void __launch_bounds__(128)
       uint32_t max_page_size = (values_in_page * 2 >= ck_g.num_values)   ? 256 * 1024
                                : (values_in_page * 3 >= ck_g.num_values) ? 384 * 1024
                                                                          : 512 * 1024;
+
+      // override max_page_size if target is smaller
+      if (max_page_size > max_page_size_bytes) max_page_size = max_page_size_bytes;
```

Comment:
We may want a better name for `max_page_size`, because it isn't clear how this value's meaning is different from `max_page_size_bytes`.
```
// TODO (dm): this convoluted logic to limit page size needs refactoring
```
@devavret Do you have suggestions on what should be done here?

Reply:
It seems to me that intent here is to prevent wildly lopsided page sizes within a column chunk.  As the target page size shrinks, this should be less necessary.  Maybe `threshold_page_size` would be a better name?  Would something like the following make sense?
```c++
auto threshold_page_size = max_page_size_bytes;
if (threshold_page_size > SOME_CUTOFF_VALUE && (values_in_page * 3 >= ck_g.num_values)) {
  threshold_page_size = (values_in_page * 2 >= ck_g.num_values) ? max_page_size_bytes << 1
                                         : (max_page_size_bytes<<2)*3;
```
`SOME_CUTOFF_VALUE` would be 512*1024 to match the current code, but could be adjusted downward if need be.

Reply:
I think that sounds reasonable but would like a second reviewer's opinion on this section. cc: @vuule @devavret

Reply:
I was thinking more on the lines of adding a condition to the if block below that says: cut off the page if the remaining size of chunk is less than max_page_size_bytes. But that would require calculating the per chunk byte size beforehand.

I think the current changes are fine for now. For naming confusion, you can rename `max_page_size` to `this_page_max_size`.

BTW I think this PR only allows the user to make the pages smaller than they were going to be. Is there any need to allow larger pages?

Reply:
> It seems to me that intent here is to prevent wildly lopsided page sizes within a column chunk.

I think the real reason may be to avoid pages with very small number of values. I think we can remove this logic altogether because the final page is going to have at least 5000 values.

Reply:
Ok, so for now I'll just rename the variable, but leave the TODO in place.